### ISSUE TITLE
[RFC] Implement the private key identification/validation

### ISSUE THREAD

**Issue by webknjaz** [sanitizers/octomachinery#28]
In the GitHub UI, it only shows a SHA-1 hash of each key. It's hard to identify which is the one the App deployment is dealing with now.
1. This value can be retrieved via OpenSSL CLI (https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#verifying-private-keys).
2. Alternatively, one could use `pycrypto`:
```python
import hashlib
from pathlib import Path

from Crypto.PublicKey import RSA


def emit_chunks(i, s): 
    c = 0 
    l = len(i) 
    while c < l: 
        e = c + s 
        yield i[c: e] 
        c = e


key_contents = Path('~/Downloads/your-app.2019-09-18.private-key.pem').expanduser().read_text()
rsa_key = RSA.importKey(key_contents)
pub_rsa_key = rsa_key.publickey()
sha1_hash = hashlib.sha1(pub_rsa_key.exportKey('DER')).hexdigest()
col_separated_sha1_hash = ':'.join(emit_chunks(sha1_hash, 2))
```
3. `pycrypto` seems to not provide wheels, can we use some other lib to avoid requiring build deps?

I think, that on the key load stage we need to store this hash value and display it in repr, optionally allowing users to provide this hash value via env var for stricter validation.

**Comment by webknjaz** [comment]
[`cryptography`]-way:
```python
from pathlib import Path
from cryptography.hazmat.backends import default_backend
from cryptography.hazmat.primitives.serialization import load_pem_private_key
from cryptography.hazmat.primitives.serialization import (
    Encoding, PrivateFormat,
    NoEncryption, PublicFormat,
)


def emit_chunks(i, s): 
    c = 0 
    l = len(i) 
    while c < l: 
        e = c + s 
        yield i[c: e] 
        c = e


key_contents = Path('~/Downloads/your-app.2019-09-18.private-key.pem').expanduser().read_text()
rsa_key = load_pem_private_key(key_contents.encode(), password=None, backend=default_backend())
pub_rsa_key = rsa_key.public_key()
sha1_hash = hashlib.sha1(pub_rsa_key.public_bytes(Encoding.DER, PublicFormat.SubjectPublicKeyInfo)).hexdigest()
col_separated_sha1_hash = ':'.join(emit_chunks(sha1_hash, 2))
```

[`cryptography`]: https://cryptography.io/

**Comment by webknjaz** [comment]
Alright, pycrypto is not an option: https://theartofmachinery.com/2017/02/02/dont_use_pycrypto.html.

TODO: check a few other libs from https://github.com/sobolevn/awesome-cryptography#python to see whether there's anything w/o sdist-only deps.

**Comment by webknjaz** [comment]
Things TODO:
* [ ] implement tests for `GitHubPrivateKey`
* [ ] implement validation/matching of the key and its SHA-1 value

**Comment by webknjaz** [comment]
Resolved by 1d66c000d195b77cf05136eba1349b237f867411 / f7a7c64845724840d01c049803972f8fd58c03e1 / 6cc66ed9cf190935fb0534fd104fc86fc904f849.
